spec/aggregator_spec.rb

```diff
@@ -51,4 +51,248 @@
       end
     end
   end
+
+  describe '#process_link' do
```

Comment:
I tried to cover every logic branch within the method, let me know if you think I've missed anything.